deploy/local_single_user_deploy.sh

```diff
@@ -0,0 +1,276 @@
+    local dest_path="$2"
+    local linked_to_lib_path="$(readlink -f ${lib_path})"
+
+    # Copy the library and maintain symbolic links recursively
+    final_dest_lib_copies=false
+    echo -e "${DEBUG}Copying ${lib_path} recursively ...${NC}"
+    while [ "$final_dest_lib_copies" != true ]
+    do
+        cp -P "$lib_path" "$dest_path/"
+        if [ "$lib_path" == "$linked_to_lib_path" ]
+        then
+            final_dest_lib_copies=true
+        fi
+        lib_path="$linked_to_lib_path"
+        linked_to_lib_path="$(readlink -f ${lib_path})"
+    done
+}
+
+copy_shared_libs() {
+    libs_visor=$(extract_shared_libraries ${VISOR_BIN})
+    libs_keeper=$(extract_shared_libraries ${KEEPER_BIN})
+    libs_client=$(extract_shared_libraries ${CLIENT_BIN})
+
+    # Combine shared libraries from all executables and remove duplicates
+    all_shared_libs=$(echo -e "${libs_visor}\n${libs_keeper}\n${libs_client}" | sort | uniq)
+
+    # Copy shared libraries to the lib directory
+    echo -e "${DEBUG}Copying shared library ...${NC}"
+    mkdir -p ${LIB_DIR}
+    for lib in ${all_shared_libs}; do
+        if [[ ! -z ${lib} ]]
+        then
+            copy_shared_libs_recursive ${lib} ${LIB_DIR}
+        fi
+    done
+}
+
+# Functions to launch and kill processes
+launch_process() {
+    local bin="$1"
+    local args="$2"
+    local log_file="$3"
+    echo -e "${DEBUG}Launching $bin ...${NC}"
+    LD_LIBRARY_PATH=${LIB_DIR} nohup ${bin} ${args} > ${BIN_DIR}/${log_file} 2>&1 &
+}
+
+kill_process() {
+    local bin="$1"
+    echo -e "${DEBUG}Killing $(basename ${bin}) ...${NC}"
+    pkill -9 -f ${bin}
+}
+
+stop_process() {
+    local bin="$1"
+    echo -e "${DEBUG}Stopping $(basename ${bin}) ...${NC}"
+    pkill -f ${bin}
+}
+
+# Main functions for install, reset, and usage
+install() {
+    echo -e "${INFO}Installing ...${NC}"
+    copy_shared_libs
+    check_files
+    generate_config_files ${NUM_KEEPERS} ${BIN_DIR} ${CONF_FILE} ${CONF_DIR}
+    launch_process ${VISOR_BIN} "--config ${CONF_DIR}/visor_conf.json" "visor.log"
+    sleep 2
+    num_keepers=${NUM_KEEPERS}
+    for (( i=0; i<num_keepers; i++ ))
+    do
+        #local keeper_args="--config ${CONF_DIR}/keeper_conf_$i.json"
+        launch_process ${KEEPER_BIN} "--config ${CONF_DIR}/keeper_conf_$i.json" "keeper_$i.log"
+    done
+    sleep 2
+    launch_process ${CLIENT_BIN} "--config ${CONF_DIR}/client_conf.json" "client.log"
+    echo -e "${DEBUG}Install done${NC}"
+}
+
+kill() {
+    echo -e "${INFO}Killing ...${NC}"
+    kill_process ${VISOR_BIN}
+    kill_process ${KEEPER_BIN}
+    kill_process ${CLIENT_BIN}
+    echo -e "${DEBUG}Kill done${NC}"
+}
+
+reset() {
+    echo -e "${INFO}Resetting...${NC}"
+    kill;
+    echo -e "${DEBUG}Delete all generated files${NC}"
+
+    # Remove all config files
+    rm ${CONF_DIR}/visor_conf.json
+    rm ${CONF_DIR}/client_conf.json
+    rm ${CONF_DIR}/keeper_conf*.json
+
+    # Remove all log files
+    rm ${BIN_DIR}/*.log
```

Comment:
this does not remove log files w/o .log extension name, such as the current *_logfile.txt. that's why I personally prefer to have .log as the extension for all log files in the conf file.